internal/pkg/client/github/client_test.go

```diff
@@ -0,0 +1,251 @@
+// -----------------------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 bomctl a Series of LF Projects, LLC
+// SPDX-FileName: internal/pkg/client/github/client_test.go
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// -----------------------------------------------------------------------------
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+package github_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/bomctl/bomctl/internal/pkg/client/github"
+	"github.com/bomctl/bomctl/internal/pkg/netutil"
+)
+
+type githubClientSuite struct {
+	suite.Suite
+}
+
+func (ghcs *githubClientSuite) TestClient_Parse() {
+	client := &github.Client{}
+
+	for _, data := range []struct {
+		expected *netutil.URL
+		name     string
+		url      string
+		owner    string
+		repoName string
+	}{
+		{
+			name:     "https scheme",
+			url:      "https://github.com/bomctl/bomctl",
+			owner:    "bomctl",
+			repoName: "bomctl",
+			expected: &netutil.URL{
+				Scheme:   "https",
+				Hostname: "github.com",
+				Path:     "bomctl/bomctl",
+			},
+		},
+		{
+			name:     "http scheme",
+			url:      "http://github.com/bomctl/bomctl",
+			owner:    "bomctl",
+			repoName: "bomctl",
+			expected: &netutil.URL{
+				Scheme:   "http",
+				Hostname: "github.com",
+				Path:     "bomctl/bomctl",
+			},
+		},
+		{
+			name:     "http scheme with tag",
+			url:      "https://github.com/ossf/scorecard",
+			owner:    "ossf",
+			repoName: "scorecard",
+			expected: &netutil.URL{
+				Scheme:   "https",
+				Hostname: "github.com",
+				Path:     "ossf/scorecard",
+			},
+		},
+		{
+			name:     "http scheme with latest tag",
+			url:      "http://github.com/bomctl/bomctl",
+			owner:    "bomctl",
+			repoName: "bomctl",
+			expected: &netutil.URL{
+				Scheme:   "http",
+				Hostname: "github.com",
+				Path:     "bomctl/bomctl",
+			},
+		},
```

Comment:
Should these have a tag value in the url?

Reply:
These were holdovers from the test file I copied it from. Removed.